### PR TITLE
Highlight primitive types

### PR DESCRIPTION
**Pull request by lnicola**
This is a hack to get better highlighting of primitive types until #1340.

### CODE REVIEW

File: crates/ra_ide_api/src/syntax_highlighting.rs [hunk 1]
```diff
@@ -61,7 +69,10 @@ pub(crate) fn highlight(db: &RootDatabase, file_id: FileId) -> Vec<HighlightedRa
             STRING | RAW_STRING | RAW_BYTE_STRING | BYTE_STRING => "string",
             ATTR => "attribute",
             NAME_REF => {
-                if let Some(name_ref) = node.as_node().and_then(ast::NameRef::cast) {
+                // FIXME: revisit this after #1340
+                if is_prim_type(&node) {
```
Comment (matklad) [inline crates/ra_ide_api/src/syntax_highlighting.rs]: It's better to use `node.text().as_str()` to avoid allocation

Comment (matklad) [reply crates/ra_ide_api/src/syntax_highlighting.rs]: That is, in the branch where we've already casted to ast::NameRef: `.text` is method on `NameRef`

Comment (lnicola) [reply crates/ra_ide_api/src/syntax_highlighting.rs]: Can we keep ignoring `SyntaxToken`s`? I'm still not sure what those are.

Comment (matklad) [reply crates/ra_ide_api/src/syntax_highlighting.rs]: `SyntaxToken` is the leaf element of  a syntax tree. Basically, a string with a tag.

`SyntaxNode` is a composite element of a syntax tree, which has other Nodes or Tokens as children

`SyntaxElement` is either of the two

Comment (lnicola) [reply crates/ra_ide_api/src/syntax_highlighting.rs]: I don't see in `SyntaxText` any `as_str()` method that doesn't allocate.

Comment (matklad) [reply crates/ra_ide_api/src/syntax_highlighting.rs]: https://github.com/rust-analyzer/rust-analyzer/blob/cb21a219987eac0f8dc08f0c858e8a7154fccd04/crates/ra_syntax/src/ast/extensions.rs#L22

Comment (lnicola) [reply crates/ra_ide_api/src/syntax_highlighting.rs]: But never mind, `==` probably works.

**Comment by lnicola** [comment]
Done, assuming CI passes.

File: crates/ra_ide_api/src/syntax_highlighting.rs [hunk 2]
```diff
@@ -30,6 +30,27 @@ fn is_control_keyword(kind: SyntaxKind) -> bool {
     }
 }
 
+fn is_prim_type(node: &ast::NameRef) -> bool {
+    let text = node.text();
+    text == "u8"
```
Comment (matklad) [inline crates/ra_ide_api/src/syntax_highlighting.rs]: This can be 

```
match node.text().as_str() {
   "i8" | "i16" ...
}
```

Comment (lnicola) [reply crates/ra_ide_api/src/syntax_highlighting.rs]: :man_facepalming:

**Comment by matklad** [comment]
bors r+

Thanks!

**Comment by lnicola** [comment]
Travis failure seems unrelated?

**Comment by matklad** [comment]
Yeah, it is unrelated. I guess we need to up the time limit on that test

**Pull request merged**